### PR TITLE
Fixes edge case where temp folder does not exist yet

### PR DESCRIPTION
**Pull request by stopthatastronaut**
Just a basic change to create the temp folder for that edge case where it doesn't exist, even though the env var does. Also swapped a "mkdir" for a "New-Item" because lint rules

### CODE REVIEW

File: OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1 [hunk 1]
```diff
@@ -389,8 +389,10 @@ function Install-Tentacle
 
     $actualTentacleDownloadUrl = Get-TentacleDownloadUrl $tentacleDownloadUrl $tentacleDownloadUrl64
 
-    mkdir "$tentacleHomeDirectory" -ErrorAction SilentlyContinue
-
+    if(-not (Test-Path $tentacleHomeDirectory))
+    {
+        New-Item -Path "$tentacleHomeDirectory" -ItemType Directory
```
Comment (matt-richardson) [inline OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1]: Wanna stick an `| out-null` on the end?

File: OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1 [hunk 2]
```diff
@@ -400,7 +402,13 @@ function Install-Tentacle
     Request-File $actualTentacleDownloadUrl $tentaclePath
 
     Write-Verbose "Installing MSI..."
-    if (-not (Test-Path "$TentacleHomeDirectory\logs")) { New-Item -type Directory "$TentacleHomeDirectory\logs" }
+    if(-not (Test-Path $env:TEMP))
+    {
+        Write-Verbose "Configured temp folder does not currently exist, creating..."
+        New-Item $env:TEMP -ItemType Directory -force # an edge case when the env var exists but the folder does not
```
Comment (matt-richardson) [inline OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1]: Wanna stick an `| out-null` on the end?

File: OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1 (hunk 1 above)
Comment (matt-richardson) [inline OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1]: Maybe a log line saying what its doing (both to be more explicit, but also to match what you're doing below?)

File: OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1 [hunk 3]
```diff
@@ -400,7 +402,13 @@ function Install-Tentacle
...
+    }
+
+    if (-not (Test-Path "$TentacleHomeDirectory\logs")) { New-Item -type Directory "$TentacleHomeDirectory\logs" -force  }
```
Comment (matt-richardson) [inline OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1]: Why do we need the force here?
If its to create the parent directory if it doesn't exist, that already happens above?

Comment (stopthatastronaut) [inline OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1]: Yeah, the force is to create all parents.
